network-api/networkapi/mozfest/templates/partials/signup.html

```diff
@@ -7,15 +7,17 @@
   <div class="sticky-cta">
     <div id="cta-anchor">
       <div class="mt-5 mt-md-0">
-        <div
-          class="join-us mb-5"
-          data-form-position="newsletter_signup_sidebar"
-          data-signup-id="{{ page.signup.id|unlocalize }}"
-          data-cta-header="{{ page.signup.header | escape }}"
-          data-cta-description="{{ page.signup.description | escape }}"
-          data-newsletter="{{ page.signup.newsletter }}"
-          data-ask-name="{{page.signup.ask_name}}">
-        </div>
+        {% with localized_signup=page.signup.localized %}
+          <div
+            class="join-us mb-5"
+            data-form-position="newsletter_signup_sidebar"
+            data-signup-id="{{ page.signup.id|unlocalize }}"
+            data-cta-header="{{ localized_signup.header | escape }}"
+            data-cta-description="{{ localized_signup.description | escape }}"
+            data-newsletter="{{ page.signup.newsletter }}"
+            data-ask-name="{{page.signup.ask_name}}">
```

Comment:
@danielfmiranda What do you think about using all field from the localized signup in the template and rather defer the decision what translated and what is synchronized to the `Signup` model? 

Looking at the current `Signup` model we are only defining `header` and `description` as translatable. The other fields are not listed at all, which leaves them in this weird place where they are only copied from the source version when the translation is first created, but are not keep in sync thereafter. I feel like this might be a source of future bugs. 

So my idea would be to add all the other signup fields as synchronized fields on the model and use all fields from the localized signup in the template. 

See also: https://www.wagtail-localize.org/how-to/field-configuration/#neither-excluded

Reply:
We might also want to figure out if `signup-id` in the HTML should be the `campaign_id` field on the model 🤔

Reply:
Hi @tbrlpld! Thanks for the link and for the explanation, this definitely makes sense to me!  

I have now updated the `SignUp` model to have the other fields it is using in the template as `SynchronizedFields`. 
I also updated all other templates in the code base that use `SignUp` to now be wrapped with the localized version of the snippet and using those localized fields instead as discussed.



To answer the second question, it appears that all instances of the signup snippets being used in HTML templates pass in the snippets own ID (unlocalized) as the `data-signup-id` variable, and not the `campaign_id`. This variable is then used in [join-us.jsx](https://github.com/mozilla/foundation.mozilla.org/blob/main/source/js/common/inject-react/join-us.js) to create a path to the foundation site's API, and not to salesforce. So I believe they are correct as is.

```jsx

    const props = element.dataset;
    const sid = props.signupId || 0;

    props.apiUrl = `${siteUrl}/api/campaign/signups/${sid}/`;

```

Reply:
Looking at `network-api/networkapi/campaign/views.py` and the functions `signup_submission_view` and `signup_submission`, I think you are correct, we should be passing the object's `id` (not the `campaign_id`). But, I think we should use the localized version. 

This again is to make sure that the differentiation between the locales is done in the `Signup` model and not in the template.

Reply:
Thanks for digging into this @danielfmiranda. It's a bit confusing in this context, but we need the `unlocalize`. This is unrelated to **which** number is displayed and only controls **how** the number is displayed. 

We use localized number display in general, e.g. "1,000" for one thousand in US style but that would be "1.000" in German number style. Either one of these would be wrong here though. Since this is a number we need to read computationally we need to the be plain "1000". That is what `unlocalize` does. 

See also: 
* https://docs.djangoproject.com/en/4.1/topics/i18n/formatting/#unlocalize

Reply:
Regarding the **which** number: I am a bit confused. `page.signup` is a synchronized field. So the french page should use the **same** signup as the english page. But `page.signup.localized` should be the translated version of the signup. Translations do have their own `id`. Did you create translations of the signup itself before these tests?